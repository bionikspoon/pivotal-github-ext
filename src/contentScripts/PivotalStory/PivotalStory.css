.PivotalStory__container {
  min-height: 44px;
}

.PivotalStory__logo {
  margin-left: -60px;
}

.PivotalStory__body {
}

.PivotalStory__name {
  height: 44px;
  font-size: 14px;
  line-height: 44px;
  padding: 0 15px;
}

.PivotalStory__id {
  color: #a3aab1;
  font-weight: 400;

  &::before {
    content: '#';
    font-weight: 400;
  }
}

.PivotalStory__description {
  padding: 15px;
  border-top: 1px solid #e1e4e8;
  font-size: 14px;
  line-height: 1.3;
}
