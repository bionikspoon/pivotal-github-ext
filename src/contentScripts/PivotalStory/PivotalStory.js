import React from 'react'
import PropTypes from 'prop-types'
import ReactMarkdown from 'react-markdown'
import { LOADING } from '../constants'

export default function PivotalStory(props) {
  if (props.data.loading !== LOADING.SUCCESS) {
    return null
  }
  return (
    <div className="PivotalStory__container">
      <h2>{props.data.story.name}</h2>
      <ReactMarkdown source={props.data.story.description} />
    </div>
  )
}

PivotalStory.propTypes = {
  data: PropTypes.shape({
    loading: PropTypes.oneOf(Object.keys(LOADING)).isRequired,
    story: PropTypes.shape({
      name: PropTypes.node.isRequired,
      description: PropTypes.node.isRequired,
    }),
  }).isRequired,
}
