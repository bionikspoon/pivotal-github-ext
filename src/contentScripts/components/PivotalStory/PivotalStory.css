.PivotalStory__container {
  min-height: 44px;
}

.PivotalStory__logo {
  margin-left: -60px;
}

.PivotalStory__body {
}

.PivotalStory__title {
  display: flex;
  min-height: 44px;
  align-items: center;
  padding: 0 15px;
  justify-content: space-between;

  & > svg {
    margin-left: 0.5em;
  }
}

.PivotalStory__name {
  font-size: 14px;
}

.PivotalStory__id {
  color: #a3aab1;
  font-weight: 400;
  margin-left: 0.5em;

  &::before {
    content: '#';
    font-weight: 400;
  }
}

.PivotalStory__description {
  padding: 15px;
  border-top: 1px solid #e1e4e8;
  font-size: 14px;
  line-height: 1.3;
  margin: 0;
}
