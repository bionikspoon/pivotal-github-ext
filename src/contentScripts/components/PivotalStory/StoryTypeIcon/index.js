/* @flow */
export { default } from './StoryTypeIcon'
