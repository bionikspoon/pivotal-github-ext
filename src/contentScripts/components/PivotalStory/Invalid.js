/* @flow */
import React from 'react'
import LoadingSpinner from '../LoadingSpinner'
import Well from '../Well'
import './PivotalStory.css'

type Props = {
  data: {
    story: {
      error: string,
      possible_fix: string,
    },
  },
}

export default function Invalid(props: Props) {
  return (
    <Well className="PivotalStory__container">
      <LoadingSpinner />
      <div className="PivotalStory__body">
        <div className="PivotalStory__title">
          <h2 className="PivotalStory__name">{props.data.story.error}</h2>
        </div>

        <p className="PivotalStory__description">
          {props.data.story.possible_fix}
        </p>
      </div>
    </Well>
  )
}
