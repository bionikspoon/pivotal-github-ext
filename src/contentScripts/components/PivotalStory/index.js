/* @flow */
import PivotalStory from './PivotalStory'
import enhancer from './enhancers'

export default enhancer(PivotalStory)
