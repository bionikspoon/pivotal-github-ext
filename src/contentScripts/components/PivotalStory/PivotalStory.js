/* @flow */
import React from 'react'
import ReactMarkdown from 'react-markdown'
import { replace } from 'ramda'
import PivotalLogo from '../PivotalLogo'
import Well from '../Well'
import './PivotalStory.css'
import StoryTypeIcon from './StoryTypeIcon'

const reUrl = /([^[(])(https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}[-a-zA-Z0-9@:%_+.~#?&//=]*)([^\])])/g
const formatLinks = replace(reUrl, '$1[$2]($2)$3')

type Props = {
  data: {
    story: {
      id: string,
      name: string,
      description: string,
      story_type: string,
    },
  },
}

export default function PivotalStory(props: Props) {
  return (
    <Well className="PivotalStory__container">
      <PivotalLogo className="PivotalStory__logo" />
      <div className="PivotalStory__body">
        <div className="PivotalStory__title">
          <h2 className="PivotalStory__name">
            {props.data.story.name}
            <span className="PivotalStory__id">{props.data.story.id}</span>
          </h2>
          <StoryTypeIcon size={20} storyType={props.data.story.story_type} />
        </div>

        <ReactMarkdown
          className="PivotalStory__description markdown-body"
          source={formatLinks(props.data.story.description)}
        />
      </div>
    </Well>
  )
}
