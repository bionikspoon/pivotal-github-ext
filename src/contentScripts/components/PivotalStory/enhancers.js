/* @flow */
import {
  applySpec,
  complement,
  compose,
  path,
  pathEq,
  pathOr,
  pipe,
  replace,
} from 'ramda'
import { branch, renderNothing, renderComponent, mapProps } from 'recompose'
import withFetch from '../../hocs/withFetch'
import withAsyncFactory from '../../../shared/utils/withAsyncFactory'
import { storage } from '../../utils/messages'
import { LOADING } from '../../../shared/constants'
import Loading from './Loading'
import Invalid from './Invalid'

const reUrl = /([^[(])(https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}[-a-zA-Z0-9@:%_+.~#?&//=]*)([^\])])/g
const formatLinks = replace(reUrl, '$1[$2]($2)$3')

export default compose(
  withAsyncFactory(() => storage({ pivotalApiKey: '' }), {
    name: 'storageData',
  }),
  branch(
    complement(pathEq(['storageData', 'loading'], LOADING.SUCCESS)),
    renderNothing
  ),
  withFetch(props => ({
    resources: {
      story: `https://www.pivotaltracker.com/services/v5/stories/${props.storyId}`,
    },
    headers: { 'X-TrackerToken': props.storageData.pivotalApiKey },
  })),
  branch(
    pathEq(['data', 'loading'], LOADING.STARTED),
    renderComponent(Loading)
  ),
  branch(path(['data', 'story', 'code', 'length']), renderComponent(Invalid)),
  branch(complement(path(['data', 'story'])), renderNothing),
  mapProps(
    applySpec({
      storyType: path(['data', 'story', 'story_type']),
      storyId: path(['data', 'story', 'id']),
      storyName: path(['data', 'story', 'name']),
      storyDescription: pipe(
        pathOr('', ['data', 'story', 'desciption']),
        formatLinks
      ),
    })
  )
)
