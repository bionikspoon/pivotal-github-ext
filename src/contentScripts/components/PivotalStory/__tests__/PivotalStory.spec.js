import React from 'react'
import { shallow } from 'enzyme'
import PivotalStory from '../PivotalStory'

describe('components/PivotalStory', () => {
  test('it renders correctly', () => {
    const props = {
      data: {
        story: {
          id: '1234',
          name: 'Create user profile page',
          description: '**expected**:\n- user can see user profile page',
          story_type: 'features',
        },
      },
    }
    const wrapper = shallow(<PivotalStory {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
