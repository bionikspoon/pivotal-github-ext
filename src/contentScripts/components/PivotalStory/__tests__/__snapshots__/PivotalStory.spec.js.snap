// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PivotalStory it renders correctly 1`] = `
<Well
  className="PivotalStory__container"
>
  <PivotalLogo
    className="PivotalStory__logo"
    fill="#ffffff"
  />
  <div
    className="PivotalStory__body"
  >
    <div
      className="PivotalStory__title"
    >
      <h2
        className="PivotalStory__name"
      >
        <a
          className="PivotalStory__link"
          href="https://www.pivotaltracker.com/story/show/1234"
        >
          Create user profile page
        </a>
        <span
          className="PivotalStory__id"
        >
          1234
        </span>
      </h2>
      <StoryTypeIcon
        fill="#5282b0"
        size={20}
        storyType="features"
      />
    </div>
    <ReactMarkdown
      className="PivotalStory__description markdown-body"
      containerTagName="div"
      parserOptions={Object {}}
      source="**expected**:
- user can see user profile page"
    />
  </div>
</Well>
`;
